### PR TITLE
refactor: Adds Plugin Docs / Reorg

### PR DESCRIPTION
**Pull request by shellscape**
This PR adds the plugin documentation from the wiki (which was recently converted to the same exact markdown format), in addition to reorganizing the documentation with a better flow as discussed in chat - use information, followed by technical information, followed by supplemental information. The reorganized docs flow such like:

```
00-introduction.md
01-command-line-reference.md
02-javascript-api.md
03-es-module-syntax.md
04-tutorial.md
05-plugins.md
06-faqs.md
07-tools.md
999-big-list-of-options.md
```

### CODE REVIEW

File: guide/en/05-plugins.md [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+transpiling non-JS files) should support `options.include` and `options.exclude`,
+both of which can be a minimatch pattern or an array of minimatch patterns. If
+`options.include` is omitted or of zero length, files should be included by
+default; otherwise they should only be included if the ID matches one of the
+patterns.
+
+The `transform` hook, if returning an object, can also include an `ast` property.
+For an example of this see
+[rollup-plugin-json](https://github.com/rollup/rollup-plugin-json/blob/master/src/index.js)
+which generates a 'fake AST' to speed up Rollup's handling of large JSON files.
```
[the remark this answers is not in the document]
Comment (shellscape) [reply guide/en/05-plugins.md]: Done

File: guide/en/05-plugins.md [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+Signature: `(code, { modules, exports, imports, fileName, isEntry }, outputOptions) => (code | { code, map} | Promise)`
+
+Can be used to transform individual chunks. Called for each Rollup output chunk
+file. Returning `null` will apply no transformations.
+
+#### `resolveId`
+Type: `Function`<br>
+Signature: `( importee, importer ) => (id|Promise)`
+
+Defines a custom resolver A resolver loader can be useful for e.g. locating
```
[the remark this answers is not in the document]
Comment (shellscape) [reply guide/en/05-plugins.md]: Done

File: guide/en/05-plugins.md [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+
+- Plugins should have a clear name with `rollup-plugin-` prefix.
+- Include `rollup-plugin` keyword in `package.json`.
+- Plugins should be tested. We recommend [mocha](https://github.com/mochajs/mocha)
+or [ava](https://github.com/avajs/ava) which support promises out of the box.
+- Use asynchronous methods when it is possible.
+- Document your plugin in English.
+- Make sure your plugin outputs correct source mappings if appropriate.
+- If your plugin uses 'virtual modules' (e.g. for helper functions), prefix the
+module ID with `\0`. This prevents other plugins from trying to process it
```
[the remark this answers is not in the document]
Comment (shellscape) [reply guide/en/05-plugins.md]: Done

File: guide/en/05-plugins.md [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+import { rollup } from 'rollup';
+import nodeResolve from 'rollup-plugin-node-resolve';
+
+rollup({
+  entry: 'main.js',
+  plugins: [ nodeResolve({ jsnext: true, main: true }) ]
+}).then(...);
+```
+
+Consult each plugin's documentation for detailed documentation.
```
[the remark this answers is not in the document]
Comment (shellscape) [reply guide/en/05-plugins.md]: Done

File: guide/en/03-es-module-syntax.md [hunk 5]
```diff
@@ -49,11 +58,15 @@ Load the module code, but don't make any new objects available.
...
-This is useful for polyfills, or when the primary purpose of the imported code is to muck about with prototypes.
+This is useful for polyfills, or when the primary purpose of the imported code
+is to muck about with prototypes.
 
 #### Dynamic Import
 
-Import modules using the [dynamic import API](https://github.com/tc39/proposal-dynamic-import#import). This API is experimental and available under the `experimentalDynamicImport` flag.
+Import modules using the
+[dynamic import API](https://github.com/tc39/proposal-dynamic-import#import).
+This API is experimental and available under the `experimentalDynamicImport`
```
[the remark this answers is not in the document]
Comment (shellscape) [reply guide/en/03-es-module-syntax.md]: Done

**Comment by lukastaegert** [comment]
Ok, I did some updates:
* merge in the master branch and resolve merge conflicts
* fix anchor links
* the plugin section used h2 headings which conflicted with the h2 headings used outside and looked weird. I shortened this section to make those headings unnecessary considering there already is an example on how to use plugins
* added an introductory example for how to write a plugin to give people a starting point and something to refer to.

The latter changes are definitely up for debate! As for the question of soft-wrapping vs. hard-wrapping: I slightly lean to the soft-wrapping option (i.e. not wrap individual lines) but can definitely live with both as long as we do not put a hard requirement on a certain line width which will require work for each update.

**Pull request merged**